lib/bolt/module_installer/specs/id/gitclone.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require_relative '../../../../bolt/module_installer/specs/id/base'
+
+module Bolt
+  class ModuleInstaller
+    class Specs
+      class ID
+        class GitClone < Base
+          # Returns the name and SHA for the module at the given ref.
+          #
+          # @param git [String] The URL to the git repo.
+          # @param ref [String] The ref to checkout.
+          # @param proxy [String] The proxy to use when cloning.
+          #
+          private_class_method def self.name_and_sha(git, ref, proxy)
+            require 'open3'
+
+            unless git?
+              Bolt::Logger.debug("'git' executable not found, unable to use git clone resolution.")
+              return nil
+            end
+
+            # Clone the repo into a temp directory that will be automatically cleaned up.
+            Dir.mktmpdir do |dir|
```

Comment:
This will cause problems for users who can't write to the system tmpdir. We might want to consider making this configurable either now or in the future. `puppetfile-resolver` supports config for doing this, so we'd be able to pass it through to that as well. We could name the config option `module-install.tmpdir` or something else.

Reply:
I don't think that's actually a problem in this case, because `Dir.mktmpdir` will try a bunch of directories until it finds one that's writable, including `.` if it has to. We run into problems when we're trying to write to a tmp dir on a _remote_ system, because we only have one option to try there.

Reply:
Oh, perfect! 👍